parsl/executors/high_throughput/process_worker_pool.py

```diff
@@ -381,14 +386,36 @@ def worker_watchdog(self, kill_event: threading.Event):
 
         logger.critical("Exiting")
 
+    @wrap_with_logs
+    def handle_monitoring_messages(self, kill_event: threading.Event):
+        """Transfer messages from the managed monitoring queue to the result queue.
+
+        We separate the queues so that the result queue does not rely on a manager
+        process, which adds overhead that causes slower queue operations but enables
+        use across processes started in fork and spawn contexts.
+
+        We transfer the messages to the result queue to reuse the ZMQ connection between
+        the manager and the interchange.
+        """
+        logger.debug("Starting monitoring messages thread")
+
+        while not kill_event.is_set():
```

Comment:
I think this loop isn't guaranteed to keep looping if there are no monitoring messages after the kill event is set (to cause monitoring_queue.get() to exit) and so setting kill_event won't always cause a termination? (eg. if there is no monitoring turned on).

So then this thread is reliant on something external killing the process worker pool, not the kill event?

```
$ pytest parsl/tests/ --config parsl/tests/configs/htex_local.py
[...]
$ grep 'Monitoring-Hand' runinfo/005/ -r
runinfo/005/htex_local/block-0/41bfdd691502/manager.log:2023-12-06 15:10:00.662 parsl:400 5911 Monitoring-Handler [DEBUG]  Starting monitoring messages thread
runinfo/005/htex_local/block-0/41bfdd691502/manager.log:2023-12-06 15:10:00.662 parsl:403 5911 Monitoring-Handler [DEBUG]  Starting monitor_queue.get
```

Reply:
Good catch. I've addressed this by adding a timeout to the `monitoring_queue.get()` call.